front/src/reducers/infra/index.ts

```diff
@@ -0,0 +1,44 @@
+import { createSlice } from '@reduxjs/toolkit';
+import type { CaseReducer, Draft, PayloadAction } from '@reduxjs/toolkit';
+import type { Feature } from 'geojson';
+
+export interface InfraStateType {
+  infraID: number | undefined;
+  featureInfoClick: { displayPopup: boolean; feature?: Feature; coordinates?: number[] };
+}
+export const infraState: InfraStateType = {
+  infraID: undefined,
+  featureInfoClick: { displayPopup: false },
+};
+
+type InfraStateReducersType<S extends InfraStateType> = {
+  ['updateInfraID']: CaseReducer<S, PayloadAction<S['infraID']>>;
+  ['updateFeatureInfoClick']: CaseReducer<S, PayloadAction<S['featureInfoClick']>>;
+};
+
+export function buildInfraStateReducers<S extends InfraStateType>(): InfraStateReducersType<S> {
+  return {
+    updateInfraID(state: Draft<S>, action: PayloadAction<S['infraID']>) {
+      state.infraID = action.payload;
+    },
+    updateFeatureInfoClick(state: Draft<S>, action: PayloadAction<S['featureInfoClick']>) {
```

Comment:
There was initially an `omit` to avoid storing an absolutely huge and unnecessary object (`_vectorTileFeature`)

```
export function updateFeatureInfoClickOSRD(featureInfoClick: OsrdConfState['featureInfoClick']) {
  return (dispatch: Dispatch) => {
    dispatch({
      type: UPDATE_FEATURE_INFO_CLICK_OSRD,
      featureInfoClick: {
        ...featureInfoClick,
        feature: omit(featureInfoClick.feature, ['_vectorTileFeature']),
      },
    });
  };
}
```

Reply:
Done